pkg/master/master.go

```diff
@@ -376,34 +361,49 @@ func logStackOnRecover(panicReason interface{}, httpWriter http.ResponseWriter)
 // init initializes master.
 func (m *Master) init(c *Config) {
 
-	nodeRESTStorage := minion.NewREST(m.minionRegistry)
+	boundPodFactory := &pod.BasicBoundPodFactory{}
+	podStorage, bindingStorage := podetcd.NewREST(c.EtcdHelper, boundPodFactory)
+	podRegistry := pod.NewRegistry(podStorage)
+
+	eventRegistry := event.NewEtcdRegistry(c.EtcdHelper, uint64(c.EventTTL.Seconds()))
+	limitRangeRegistry := limitrange.NewEtcdRegistry(c.EtcdHelper)
+	resourceQuotaRegistry := resourcequota.NewEtcdRegistry(c.EtcdHelper)
+	m.namespaceRegistry = namespace.NewEtcdRegistry(c.EtcdHelper)
+
+	// TODO: split me up into distinct storage registries
+	registry := etcd.NewRegistry(c.EtcdHelper, podRegistry)
+
+	m.serviceRegistry = registry
+	m.endpointRegistry = registry
+	m.nodeRegistry = registry
+
+	nodeStorage := minion.NewREST(registry)
 	podCache := NewPodCache(
 		c.KubeletClient,
-		RESTStorageToNodes(nodeRESTStorage).Nodes(),
-		m.podRegistry,
+		RESTStorageToNodes(nodeStorage).Nodes(),
+		podRegistry,
 	)
 	go util.Forever(func() { podCache.UpdateAllContainers() }, m.cacheTimeout)
 	go util.Forever(func() { podCache.GarbageCollectPodStatus() }, time.Minute*30)
 
+	// TODO: refactor podCache to sit on top of podStorage via status calls
```

Comment:
Actually, my goal is to entirely nuke the current podCache once all components send their status to the apiserver.

I expect we'll eventually need a different kind of cache. I'd like a generic cache in either apiserver or the registry to keep all objects up to date in memory. GET should never hit etcd directly. But I'm not sure whether we need this yet.